### PR TITLE
[mp] New dbt quickstart

### PR DESCRIPTION
**Pull request by mattppal**
# Description

Adds a new dbt quickstart with two pipelines:

1. Simple dbt pipeline— designed to get users running as fast as possible. Uses the `jaffle_shop` demo to extract, transform, and load data with a combination of dbt & python.
2. External repo demo— pulls an _external_ dbt repo and executes it... In the same pipeline.

This is built using docker compose and designed to be executed in under a minute.

Staging: https://mage-staging.mintlify.app/guides/setup-dbt

https://github.com/mage-ai/mage-ai/assets/59450879/71a3a29b-c67f-4d38-b225-89a9af95ac27

# How Has This Been Tested?

- [x] Mintlify dev
- [x] Mintlify broken-links
- [x] Run on MBP Intel from cold start
- [x] Run on MBP M1 from cold start

# Checklist
- [x] The PR is tagged with proper labels (bug, enhancement, feature, documentation)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: docs/guides/setup-dbt.mdx
Comment (johnson-mage) [inline docs/guides/setup-dbt.mdx:51]: I think this could be clearer by saying "double-clicking the row from the list of pipelines".

File: docs/guides/setup-dbt.mdx
Comment (johnson-mage) [inline docs/guides/setup-dbt.mdx:83]: I think this could be clearer by saying "double-clicking the row from the list of pipelines".

File: docs/guides/setup-dbt.mdx
Comment (johnson-mage) [inline docs/guides/setup-dbt.mdx:52]: Perhaps rephrase to `then select "Edit pipeline" (with the "code" icon) from the side nav`. Just referring to "code icon" might not be super obvious to some.

File: docs/guides/setup-dbt.mdx
Comment (johnson-mage) [reply docs/guides/setup-dbt.mdx:83]: Perhaps rephrase `then select the "code" icon from the side nav.` to `then select "Edit pipeline" (with the "code" icon) from the side nav`.

**Comment by mattppal** [comment]
@johnson-mage Thanks, I'll hold off until that PR is merged

**Comment by johnson-mage** [comment]
@mattppal There were a lot of files added after resolving the merge conflict that seem like they should not be included (e.g. files in the `frontend_dist` and `frontend_dist_base_path_template` directories and a bunch of files that seem like duplicates and have an extra ` 2` appended to the filename. Can you please double check?

**Comment by johnson-mage** [comment]
@mattppal In terms of the docs updated in https://github.com/mage-ai/mage-ai/pull/3497, it didn't seem like a lot so I wouldn't have expected there to be a ton of merge conflicts.

**Comment by mattppal** [comment]
@johnson-mage thanks for the heads up, that's weird. I'll just remove anything not in `docs`

**Comment by mattppal** [comment]
@johnson-mage ok that was a git nightmare, but we should be good. Thanks for your eyes!

File: docs/guides/setup-dbt.mdx
Comment (johnson-mage) [reply docs/guides/setup-dbt.mdx:51]: @mattppal Was this addressed to clarify the language? Or you don't think it's necessary?

File: docs/guides/setup-dbt.mdx
Comment (johnson-mage) [reply docs/guides/setup-dbt.mdx:52]: Same here

**Comment by johnson-mage** [comment]
@mattppal Right now, the only doc i see updated is `setup-dbt.mdx`. In the previous commits, I saw some other docs updated. Are we missing any updated docs? For example, this screenshot was taken from a previous commit and shows a few other .mdx files updated or added.
image

**Comment by wangxiaoyou1993** [comment]
We shouldn't delete the old guide. The new quickstart is good for users to run a dbt pipeline inside Mage. But it doesn't show user how to create a new dbt project inside Mage, how to feed the data of a python block into dbt block, etc.

**Comment by mattppal** [comment]
> @mattppal Right now, the only doc i see updated is `setup-dbt.mdx`. In the previous commits, I saw some other docs updated. Are we missing any updated docs? For example, this screenshot was taken from a previous commit and shows a few other .mdx files updated or added. image

Those were tangential and merged in via another update :) 

e.g. https://docs.mage.ai/development/updating-mage

**Comment by mattppal** [comment]
> We shouldn't delete the old guide. The new quickstart is good for users to run a dbt pipeline inside Mage. But it doesn't show user how to create a new dbt project inside Mage, how to feed the data of a python block into dbt block, how to write tests, etc.

Good call, I'll move it to a separate guide.

File: docs/guides/setup-dbt.mdx
Comment (mattppal) [reply docs/guides/setup-dbt.mdx:51]: I changed it, but it must have gotten overwritten with the git stuff. Nice catch!

**Comment by mattppal** [comment]
@wangxiaoyou1993 previous guide added

File: docs/guides/setup-dbt.mdx
Comment (johnson-mage) [reply docs/guides/setup-dbt.mdx:52]: nit: run-on sentence. Period needed after "list of pipelines".

Comment (johnson-mage) [reply docs/guides/setup-dbt.mdx:52]: > Perhaps rephrase to `then select "Edit pipeline" (with the "code" icon) from the side nav`. Just referring to "code icon" might not be super obvious to some.

this change mightve gotten overwritten

File: docs/guides/setup-dbt.mdx
Comment (johnson-mage) [inline docs/guides/setup-dbt.mdx:83]: > Perhaps rephrase to `then select "Edit pipeline" (with the "code" icon) from the side nav`. Just referring to "code icon" might not be super obvious to some.

this change mightve gotten overwritten

**Comment by johnson-mage** [comment]
shud be good after moving the gifs/images over to the assets repo

**Review by jamesking-github — APPROVED** [review]
I think this functionality is already in the tool, as someone experiencing issues running dbt within mage updated documentation is appreciated

**Review by johnson-mage — APPROVED** [review]

**Pull request merged**
